src/kakarot/state.cairo

```diff
@@ -435,17 +439,41 @@ namespace Internals {
     //      Kakarot is not authorized for accounts that are created and SELDESTRUCT in the same transaction
     // @param transfers_len The length of the transfers array.
     // @param transfers The array of Transfer.
-    func _transfer_eth{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
-        token_address: felt, transfers_len: felt, transfers: model.Transfer*
-    ) {
+    func _transfer_eth{
+        syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, accounts: DictAccess*
+    }(token_address: felt, transfers_len: felt, transfers: model.Transfer*) {
         if (transfers_len == 0) {
             return ();
         }
 
         let transfer = [transfers];
-        IERC20.transferFrom(
-            token_address, transfer.sender.starknet, transfer.recipient.starknet, transfer.amount
-        );
+
+        let (kakarot_address) = get_contract_address();
+
+        let (pointer) = dict_read{dict_ptr=accounts}(key=transfer.recipient.starknet);
+        let recipient = cast(pointer, model.Account*);
```

Comment:
we should probably add some extensive comments to explain what is being done here,
just like the PR description